src/annotator/anchoring/range-js.js

```diff
@@ -32,3 +34,51 @@ export function nodeFromXPath(xpath, root) {
   });
   return node;
 }
+
+/**
+ * Attempt to evaluate a provided xpath. If the evaluation fails, then fall back to a
+ * manual evaluation using `nodeFromXPathFallback` that can construct a simple xpath
+ * that is somewhat reasonable, albeit perhaps incomplete or less accurate.
```

Comment:
```suggestion
 * manual evaluation using `nodeFromXPathFallback` that can evaluate very simple XPaths such
 * as those generated by `xpathFromNode`
```

This function evaluates rather than constructs XPaths. I think it is also helpful to define what kind of XPath it can handle in the fallback path with reference to `xpathFromNode`.